webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx

```diff
@@ -197,25 +200,29 @@ function ResultDetails() {
       ],
       isDarkMode,
     );
-  }, [matrixRes.data, isDarkMode, resultColHeaders]);
+  }, [matrixRes.data, resultColHeaders, isDarkMode]);
 
   ////////////////////////////////////////////////////////////////
   // Event Handlers
   ////////////////////////////////////////////////////////////////
 
-  const handleItemTypeChange: ToggleButtonGroupProps["onChange"] = (
-    _,
-    newValue: OutputItemType,
-  ) => {
-    if (newValue && newValue !== itemType) {
-      setItemType(newValue);
-    }
-  };
+  const handleItemTypeChange: ToggleButtonGroupProps["onChange"] = useCallback(
```

Comment:
Give the type to the hook useCallback, so the type of the function argument will infer automatically:

```typescript
const handleItemTypeChange = useCallback<ToggleButtonGroupProps["onChange"]>((_event, newValue) => ...);
```